[Help Request] Unable to complete installation according to the document.

### High Level Description

I followed the doc to install SMARTS but failed. There are 2 problems:
1. `$ pip install -e .[camera_obs,sumo,test]`  should be `$ pip install -e '.[camera_obs,sumo,test]'`
2.  Fail to execute `$ make sanity-test`, error info:

(.venv) ➜  SMARTS git:(v1.4.0) make sanity-test
scl scenario build --clean scenarios/sumo/loop
Building: scenarios/sumo/loop
Error processing line 1 of /home/XX/SMARTS/.venv/lib/python3.8/site-packages/__editable__.smarts-1.4.0.pth:

Fatal Python error: init_import_size: Failed to import the site module
Python runtime state: initialized
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site.py", line 169, in addpackage
    exec(line)
  File "<string>", line 1, in <module>
  File "/home/XX/SMARTS/.venv/lib/python3.8/site-packages/__editable___smarts_1_4_0_finder.py", line 2, in <module>
    from importlib.machinery import ModuleSpec, PathFinder
  File "/usr/local/lib/python3.8/importlib/__init__.py", line 57, in <module>
    import types
  File "/home/XX/SMARTS/smarts/sstudio/types/__init__.py", line 20, in <module>
    from smarts.sstudio.types.actor.social_agent_actor import *
ModuleNotFoundError: No module named 'smarts'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site.py", line 580, in <module>
    main()
  File "/usr/local/lib/python3.8/site.py", line 563, in main
    known_paths = venv(known_paths)
  File "/usr/local/lib/python3.8/site.py", line 495, in venv
    addsitepackages(known_paths, [sys.prefix])
  File "/usr/local/lib/python3.8/site.py", line 350, in addsitepackages
    addsitedir(sitedir, known_paths)
  File "/usr/local/lib/python3.8/site.py", line 208, in addsitedir
    addpackage(sitedir, name, known_paths)
  File "/usr/local/lib/python3.8/site.py", line 179, in addpackage
    import traceback
  File "/usr/local/lib/python3.8/traceback.py", line 5, in <module>
    import linecache
  File "/usr/local/lib/python3.8/linecache.py", line 11, in <module>
    import tokenize
  File "/usr/local/lib/python3.8/tokenize.py", line 32, in <module>
    import re
  File "/usr/local/lib/python3.8/re.py", line 124, in <module>
    import enum
  File "/usr/local/lib/python3.8/enum.py", line 2, in <module>
    from types import MappingProxyType, DynamicClassAttribute
  File "/home/XX/SMARTS/smarts/sstudio/types/__init__.py", line 20, in <module>
    from smarts.sstudio.types.actor.social_agent_actor import *
ModuleNotFoundError: No module named 'smarts'
Command '['/home/XX/SMARTS/.venv/bin/python3.8', 'scenario_builder.py', '/home/XX/SMARTS/scenarios/sumo/loop/scenario.py', '42']' returned non-zero exit status 1.
Makefile:87: recipe for target 'build-sanity-scenarios' failed
make: *** [build-sanity-scenarios] Error 1

### Version

1.4.0

### Operating System

ubuntu 18.04

### Problems

_No response_

--- a/docs/setup.rst
+++ b/docs/setup.rst
@@ -15,11 +15,31 @@ Demo Video
 Prerequisites
 -------------
 
-+ ``python3 (3.8 or 3.9)``
++ ``python3 (3.8, 3.9, 3.10, 3.11)``
 + ``ubuntu (>=16.04)``
 
 Installation
 ------------
+
+Package
+^^^^^^^
+
+This includes SMARTS but none of the examples.
+
+.. code-block:: bash
+
+    # For Mac OS X users, ensure XQuartz is pre-installed.
+    # Install the system requirements. You may use the `-y` option to enable
+    # automatic assumption of "yes" to all prompts to avoid timeout from 
+    # waiting for user input. 
+    $ bash utils/setup/install_deps.sh
+
+    # This should install the latest version of SMARTS from package index (generally PyPI).
+    $ pip install 'smarts[camera_obs,sumo,example]'
+
+
+Development
+^^^^^^^^^^^
 
 Run the following commands to setup the SMARTS simulator.
 
@@ -61,9 +81,11 @@ Run the following commands to setup the SMARTS simulator.
     # Install smarts with extras as needed. Extras include the following: 
     # `camera_obs` - needed for rendering camera observations, and for testing.
     # `sumo` - needed for using SUMO scenarios.
-    # `test` - needed for testing.
+    # `test` - needed for running tests.
     # `example` - needed for running examples.
-    $ pip install -e .[camera_obs,sumo,test,example]
+    # `--config-settings editable_mode=strict` - may be needed depending on version of setuptools. 
+    #      See https://github.com/huawei-noah/SMARTS/issues/2090.
+    $ pip install -e '.[camera_obs,sumo,test,example]' --config-settings editable_mode=strict
 
     # Run sanity-test and verify they are passing.
     # If tests fail, check './sanity_test_result.xml' for test report. 

